logbus/solvermon/vertexmon.go

```diff
@@ -37,111 +37,155 @@ type vertexMonitor struct {
 	isCanceled     bool
 }
 
-var reErrExitCode = regexp.MustCompile(`^(?:process ".*" did not complete successfully|error calling LocalhostExec): exit code: (?P<exit_code>[0-9]+)$`)
-var reErrNotFound = regexp.MustCompile(`^failed to calculate checksum of ref ([^ ]*): (.*)$`)
-var reHint = regexp.MustCompile(`^(?P<msg>.+?):Hint: .+`)
+var reErrExitCode = regexp.MustCompile(`(?:process ".*" did not complete successfully|error calling LocalhostExec): exit code: (?P<exit_code>[0-9]+)$`)
 
-func (vm *vertexMonitor) Write(dt []byte, ts time.Time, stream int) (int, error) {
-	if stream == BuildkitStatsStream {
-		stats, err := vm.ssp.Parse(dt)
+func getExitCode(errString string) (int, bool) {
+	if matches, _ := stringutil.NamedGroupMatches(errString, reErrExitCode); len(matches["exit_code"]) == 1 {
+		exitCodeMatch := matches["exit_code"][0]
+		// exit codes should be max 255 (8 bits) however determineFatalErrorType
+		// expects math.MaxUint32 to be a special case for OOM, so we allow it here
+		exitCode, err := strconv.ParseInt(exitCodeMatch, 10, 64)
 		if err != nil {
-			return 0, errors.Wrap(err, "failed decoding stats stream")
+			return 0, false
 		}
-		for _, statsSample := range stats {
-			statsJSON, err := json.Marshal(statsSample)
-			if err != nil {
-				return 0, errors.Wrap(err, "stats json encode failed")
-			}
-			_, err = vm.cp.Write(statsJSON, ts, int32(stream))
-			if err != nil {
-				return 0, errors.Wrap(err, "write stats")
-			}
+		// Check if the exit code can fit into an int
+		if exitCode > int64(^uint(0)>>1) {
```

Comment:
To have getExitCode handle the existing case of math.MaxUint32 being used to indicate OOM and not get caught by the "go/incorrect-integer-conversion" lint rule, need to handle the case that exit code is greater than MaxInt